src/block/BigDripleaf.php

```diff
@@ -0,0 +1,233 @@
+			if($next->isHead()){
+				return $next;
+			}
+			$step++;
+		}
+		return null;
+	}
+
+	private function grow(?Player $player) : bool{
+		$head = $this->seekToHead();
+		if($head === null){
+			return false;
+		}
+		$pos = $head->getPosition();
+		$up = $pos->up();
+		$world = $pos->getWorld();
+		if(
+			!$world->isInWorld($up->getFloorX(), $up->getFloorY(), $up->getFloorZ()) ||
+			$world->getBlock($up)->getTypeId() !== BlockTypeIds::AIR
+		){
+			return false;
+		}
+
+		$tx = new BlockTransaction($world);
+
+		$tx->addBlock($pos, (clone $head)->setHead(false));
+		$tx->addBlock($up, VanillaBlocks::BIG_DRIPLEAF()
+			->setFacing($head->getFacing())
+			->setHead(true)
+		);
+
+		$ev = new StructureGrowEvent($head, $tx, $player);
+		$ev->call();
+
+		if(!$ev->isCancelled()){
+			return $tx->apply();
+		}
+		return false;
+	}
+
+	private function setTiltAndScheduleTick(DripleafTiltType $tilt) : void{
+		$this->position->getWorld()->setBlock($this->position, $this->setTilt($tilt));
+		$this->position->getWorld()->scheduleDelayedBlockUpdate($this->position, $tilt->getUpdateTicks());
+	}
+
+	public function onEntityInside(Entity $entity) : bool{
+		if(!$entity instanceof Living || !$this->head || !$this->tilt->equals(DripleafTiltType::NONE())){
+			return false;
+		}
+		$this->setTiltAndScheduleTick(DripleafTiltType::UNSTABLE());
+		return true;
+	}
+
+	public function onProjectileHit(Projectile $projectile, RayTraceResult $hitResult) : void{
+		if($this->head){
+			$this->setTiltAndScheduleTick(DripleafTiltType::FULL());
+			$this->position->getWorld()->addSound($this->position, new DripleafTiltDownSound());
+		}
+	}
+
+	public function onScheduledUpdate() : void{
+		if($this->head && !$this->tilt->equals(DripleafTiltType::NONE())){
+			if($this->tilt->equals(DripleafTiltType::FULL())){
+				$this->position->getWorld()->setBlock($this->position, $this->setTilt(DripleafTiltType::NONE()));
+				$this->position->getWorld()->addSound($this->position, new DripleafTiltUpSound());
+			}else{
+				$this->setTiltAndScheduleTick(match($this->tilt->id()) {
+					DripleafTiltType::UNSTABLE()->id() => DripleafTiltType::PARTIAL(),
+					DripleafTiltType::PARTIAL()->id() => DripleafTiltType::FULL(),
+					default => throw new AssumptionFailedError("All types should be covered")
+				});
+				$this->position->getWorld()->addSound($this->position, new DripleafTiltDownSound());
+			}
+		}
+	}
+
+	protected function recalculateCollisionBoxes() : array{
+		if(!$this->head || $this->tilt->equals(DripleafTiltType::FULL())){
+			return [];
+		}
+		return [
+			AxisAlignedBB::one()
+			->trim(Facing::DOWN, 11 / 16)
+			->trim(Facing::UP, match($this->tilt->id()) {
+				DripleafTiltType::NONE()->id(), DripleafTiltType::UNSTABLE()->id() => 1 / 16,
+				DripleafTiltType::PARTIAL()->id() => 3 / 16,
+				default => throw new AssumptionFailedError("All types should be covered")
+			})
+		];
+	}
+
+	public function isFlammable() : bool{
+		return true;
+	}
```

Comment:
The flame encouragement and flammability values should be set instead of overriding this - this is not supposed to be overridden

A Bedrock data dump suggests that the flammability is `100` and the flame encouragement is `15`